### PR TITLE
Storage docs update for requester pays and user project

### LINKED ISSUE #1773: [Request] Verify User Project methods comments are not ambiguous
General Request:

Verify that the comments are clear for methods which accept User Project with and without Requester Pays.

Here's a distinction between Requester Pays and User Project:
User Project: Is a request metadata field that is used to assign Storage operational costs to a specified project.

Requester Pays: Is a bucket-level metadata field that is used to enforce the use of the User Project metadata field to assign operational costs when an operation is made on a Bucket and its objects.

### PR DESCRIPTION
**Pull request by quartzmo**
* Remove whitelist warning.
* Remove mention of requester pays bucket from user_project docs.

[closes #1773]

### CODE REVIEW

File: google-cloud-storage/lib/google/cloud/storage/bucket.rb [hunk 1]
```diff
@@ -48,23 +48,19 @@ class Bucket
...
-        # transit costs will be billed to the that project. The default is
-        # `nil`.
+        # A boolean value or a project ID string to indicate the project to
+        # be billed for operations on the bucket and its files. If this
+        # attribute is set to `true`, transit costs for operations on the bucket
+        # will be billed to the current project for this client. (See
+        # {Project#project} for the ID of the current project.) If this
+        # attribute is set to a project ID, and that project is authorized for
+        # the currently authenticated service account, transit costs will be
+        # billed to the that project. The default is `nil`.
```
Comment (frankyn) [inline google-cloud-storage/lib/google/cloud/storage/bucket.rb]: This is great, what do you think about adding an additional line for?

The parameter `user_project` is required with requester pays enabled bucket.

File: google-cloud-storage/lib/google/cloud/storage/project.rb [hunk 2]
```diff
@@ -181,7 +181,7 @@ def buckets prefix: nil, token: nil, max: nil, user_project: nil
         #   bucket = storage.bucket "my-bucket"
         #   puts bucket.name
         #
-        # @example With `user_project` set to pay for a requester pays bucket:
+        # @example With `user_project` set to the default project:
```
[the remark this answers is not in the document]
Comment (quartzmo) [reply google-cloud-storage/lib/google/cloud/storage/project.rb]: Sadly that doesn't fit. (YARD example titles must stay on one line and not wrap, afaik). What about: "With `user_project` set to bill costs to the default project:"

File: google-cloud-storage/lib/google/cloud/storage/bucket.rb (hunk 1 above)
Comment (quartzmo) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: SGTM, I'll add this everywhere these `user_project` docs appear.

**Comment by quartzmo** [comment]
@frankyn Updated.

File: google-cloud-storage/lib/google/cloud/storage/project.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (frankyn) [reply google-cloud-storage/lib/google/cloud/storage/project.rb]: Ah okay! The suggested line LGTM!

**Pull request merged**